### PR TITLE
Restrict tagging platform users who are not member of Space

### PR DESCRIPTION
**Pull request by hero101**
> The list includes both Users and VCs.
>  
> Space L0 Public
> All Users + VCs are mentionable
> Reasoning: Every registered user has access to the Space and it;s contents
>  
> Space L0 Private
> Only members of the L0 are mentionable - Users + VCs
> Reasoning: The Subspace contents are restricted and can only be accessed by members of the Subspace.
>  
> Space L0 Public + Subspace L1 Public
> All Users + VCs are mentionable
> Reasoning: Every registered user has access to the Space and it;s contents L0 + L1
>  
> Space L0 Public + Subspace L1 Private
> Only members of the L1 are mentionable - Users + VCs
> Reasoning: The Subspace contents are restricted and can only be accessed by members of the Subspace.
>  
> Space L0 Private + Subspace L1 Public
> Only members of the Space L0 can be mentioned - Users + VCs
> Reasoning: contents are visible to everyone who has access to the Space L0
>  
> Space L0 Private + Subspace L1 Private
> Only members of the L1 are mentionable - Users + VCs
> Reasoning: The Subspace contents are restricted and can only be accessed by members of the Subspace.

## Summary by CodeRabbit

* **New Features**
  * Added mentionable contributors for spaces: search/typeahead by name, optional actor-type filtering, and configurable result limits; respects space privacy and parent-space hierarchy.
* **Tests**
  * Added unit tests covering mentionable contributor lookup, scope resolution across space hierarchies, filtering behavior, limits clamping, and resolver integration.

[Review Change Stack](https://app.coderabbit.ai/change-stack/alkem-io/server/pull/6078?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/domain/actor/actor-lookup/actor.lookup.service.ts [hunk 1]
```diff
@@ -372,6 +374,75 @@ export class ActorLookupService {
...
+   * @param limit            Hard cap on returned rows (typeahead UX). Defaults to 25.
+   */
+  async findMentionableContributors(
+    scope:
+      | { allPlatform: true }
+      | { allPlatform: false; credentials: ICredentialDefinition[] },
+    actorTypes: ActorType[] = [ActorType.USER, ActorType.VIRTUAL_CONTRIBUTOR],
+    filter?: ContributorFilterInput,
+    limit: number = 25
+  ): Promise<IActorFull[]> {
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/domain/actor/actor-lookup/actor.lookup.service.ts:395]: Applied — empty `actorTypes` now falls back to the default pair (prevents `IN ()` SQL error) and `limit` is clamped to `[1, 50]`. Tests added for both edge cases. Commit: cfdb708b2.

File: src/domain/actor/actor-lookup/actor.lookup.service.ts [hunk 2]
```diff
@@ -372,6 +374,75 @@ export class ActorLookupService {
...
+        return [];
+      }
+      // EXISTS subquery avoids row duplication that an INNER JOIN on credentials
+      // would produce when an Actor matches multiple ancestor member credentials.
+      qb.andWhere(qb2 => {
+        const sub = qb2
+          .subQuery()
+          .select('1')
+          .from(Credential, 'cred')
+          .where('cred.actorId = actor.id')
```
[the remark this answers is not in the document]
Comment (hero101) [reply src/domain/actor/actor-lookup/actor.lookup.service.ts:423]: False positive. TypeORM's `replacePropertyNames` (`QueryBuilder.js:492-502`) registers BOTH `column.databaseName` AND `column.propertyName` as keys in the replacements map, each mapping to the column's database name. So `cred.actorId` and `cred.actorID` both resolve to the escaped column `"cred"."actorId"`. No runtime failure. Leaving as-is.

**Review by bobbykolev — APPROVED** [review]
